std/math/algebraic.d

```diff
@@ -43,24 +43,23 @@ import std.traits : CommonType, isFloatingPoint, isIntegral, isSigned, Unqual;
  *     the return type will be the same as the input.
  *
  * Limitations:
- *     Does not work correctly for signed intergal types and value `Num`.min.
+ *     Does not work correctly for value `Num`.min.
```

Comment:
It should also say more clearly what it actually does rather than just "doesn't work correctly" — in fact, D has defined semantics here so is it wrong?

Reply:
I'm not sure I understand what behaviour of `abs` you expect in this case.
"Calculates the absolute value of a number" is not representable in the case of `int.min`. From where I'm standing this is currently undefined. Do we expect saturation, zero, identity? (`int.max` / `0` / `int.min`)
Do we even want to define behaviour for `abs(Num.min)` at all?

Under the current code the return value of `Num.min`, in the case of signed integral types, will be itself (identity), as `-Num.min` will return itself (overflow). I view this more as an implementation side-effect.

I don't believe this should be defined behaviour.
As an example, this is what cpp's (int/long/long long) `std::abs` states:
> The behavior is undefined if the result cannot be represented by the return type.

On a side-tangent, I've searched for the specifications of unary `-` before, but I believe these have been implicitly borrowed from `C(++)`. I've been unable to locate them in the documentation. This goes for both `-Num.min` and `-unsigned`: https://discord.com/channels/242094594181955585/242122752436338688/1169211488062406686 .
This should definitely be explicitly defined behaviour.